src/Controller/ExportController.php

```diff
@@ -128,9 +128,43 @@ private function fillDataFromResponse(array &$data, array $response): array
      */
     private function csv(array $rows, string $objectType): void
     {
-        // create csv string data
-        $fields = array_shift($rows);
+        $result = $this->processCsv($rows, $objectType);
+        extract($result); // => $filename, $csv
+        $this->outputCsv($filename, $csv);
+    }
+
+    /**
+     * Output csv file.
+     *
+     * @param string $filename The file name.
+     * @param string $csv The csv string data.
+     * @return void
+     */
+    protected function outputCsv($filename, $csv): void
+    {
+        header("Content-type: text/csv;charset=UTF-8");
```

Comment:
We should probably use https://book.cakephp.org/3/en/controllers/request-response.html#sending-files 
With some additional headers probably... 
This way we should avoid to call `exit` and create mock